AdyenComponents/BACS Direct Debit/BACSDirectDebitComponent.swift

```diff
@@ -85,6 +68,24 @@ public final class BACSDirectDebitComponent: PaymentComponent, PresentableCompon
         self.bacsPaymentMethod = paymentMethod
         self.apiContext = apiContext
         self.configuration = configuration
+        self.inputFormViewController = BACSInputFormViewController(title: paymentMethod.name,
+                                                                   styleProvider: configuration.style)
+        self.inputPresenter = createInputPresenter()
+        inputFormViewController.presenter = inputPresenter
+    }
+    
+    private func createInputPresenter() -> BACSInputPresenterProtocol {
+        let tracker = BACSDirectDebitComponentTracker(paymentMethod: bacsPaymentMethod,
+                                                      apiContext: apiContext,
+                                                      isDropIn: _isDropIn)
+        let itemsFactory = BACSItemsFactory(styleProvider: configuration.style,
+                                            localizationParameters: configuration.localizationParameters,
+                                            scope: String(describing: self))
+        return BACSInputPresenter(view: inputFormViewController,
```

Comment:
The reason we had to make those lazy was due to payment not being set on init but after init for the components. I believe when you call `createInputPresenter()` in init, `payment.amount` will be nil :). Did you confirm it?

Reply:
good that you mentioned that! I'll look into it.

the problem was that both lazy var's refer to each other and we end up with an infinite recursion :-).